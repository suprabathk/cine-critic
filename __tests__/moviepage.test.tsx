import MoviePage from "@/pages/movie/[movieID]";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const dummyTitle = "12345678901234567890123456789012345678901";
const dummyDescription =
  "123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901";

afterEach(() => {
  jest.clearAllMocks();
});

jest.mock("../src/utils/auth-utils", () => ({
  isAuth: jest.fn().mockReturnValue(true),
  getUsername: jest.fn().mockReturnValue("reviewUsername"),
}));

jest.mock("../src/hooks/useReviewsQuery", () => ({
  useReviewsQuery: jest.fn().mockReturnValue({
    data: [
      {
        id: "1",
        title: "reviewTitle",
        description: "reviewDescription",
        rating: 0,
        username: "reviewUsername",
      },
    ],
    isLoading: false,
    error: {},
  }),
}));

jest.mock("../src/hooks/useMovieQuery", () => ({
  useMovieQuery: jest.fn().mockReturnValue({
    data: {
      adult: false,
      backdrop_path: "",
      genre_ids: [],
      id: "1",
      original_language: "",
      original_title: "",
      overview: "overview",
      popularity: 0,
      poster_path: "",
      release_date: "release",
      title: "title",
      vote_average: 0,
      vote_count: 0,
      genres: [],
      production_companies: [],
      revenue: 0,
      budget: 0,
      runtime: 0,
      tagline: "tagline",
    },
    isLoading: false,
    error: {},
  }),
}));

const mutationFn = jest.fn();
jest.mock("@tanstack/react-query", () => ({
  ...jest.requireActual("@tanstack/react-query"),
  useQueryClient: () => ({
    invalidateQueries: jest.fn(),
  }),
  useMutation: () => ({
    mutate: mutationFn,
  }),
}));

const pushMock = jest.fn();
jest.mock("next/router", () => ({
  useRouter() {
    return {
      push: pushMock,
      query: { movieID: "1" },
    };
  },
}));

describe("Moviepage tests", () => {
  it("Renders reviews", async () => {
    render(<MoviePage />);
    await waitFor(() => {
      const heading = screen.getByRole("heading", {
        name: /reviewTitle/i,
      });
      const description = screen.getByText(/reviewDescription/i);
      const username = screen.getByText(/reviewUsername/i);
      expect(heading).toBeInTheDocument();
      expect(description).toBeInTheDocument();
      expect(username).toBeInTheDocument();
    });
  });

  describe("Add review tests", () => {
    it("Add review modal", async () => {
      render(<MoviePage />);
      const addReviewButton = screen.getByRole("button", {
        name: /add review/i,
      });
      await userEvent.click(addReviewButton);
      await waitFor(() => {
        const addReviewTitle = screen.getByRole("heading", {
          name: /add review/i,
        });
        expect(addReviewTitle).toBeInTheDocument();
      });
    });

    it("Submit add review", async () => {
      render(<MoviePage />);
      const addReviewButton = screen.getByRole("button", {
        name: /add review/i,
      });
      await userEvent.click(addReviewButton);
      const reviewTitle = await screen.findByRole("textbox", {
        name: /title/i,
      });
      const reviewDescription = await screen.findByRole("textbox", {
        name: /description/i,
      });
      const reviewRating = await screen.findByLabelText("star-1");
      const submitButton = screen.getByRole("button", {
        name: /submit review/i,
      });

      await userEvent.type(reviewTitle, "1234");
      await userEvent.type(reviewDescription, "1234");
      await userEvent.click(reviewRating);
      await userEvent.click(submitButton);

      await waitFor(() => {
        expect(mutationFn).toBeCalledTimes(1);
      });
    });

    describe("Rating input tests", () => {
      it("Rating is required", async () => {
        render(<MoviePage />);
        const addReviewButton = screen.getByRole("button", {
          name: /add review/i,
        });
        await userEvent.click(addReviewButton);

        expect(
          screen.queryByRole("heading", {
            name: /rating is required/i,
          })
        ).toBeNull();

        const submitButton = screen.getByRole("button", {
          name: /submit review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(
            screen.getByRole("heading", {
              name: /rating is required/i,
            })
          ).toBeInTheDocument();
        });
      });
    });

    describe("Title input tests", () => {
      it("Title is required", async () => {
        render(<MoviePage />);
        const addReviewButton = screen.getByRole("button", {
          name: /add review/i,
        });
        await userEvent.click(addReviewButton);

        expect(screen.queryByText(/title is required/i)).toBeNull();

        const submitButton = screen.getByRole("button", {
          name: /submit review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(screen.getByText(/title is required/i)).toBeInTheDocument();
        });
      });

      it("Title cannot be longer than 40", async () => {
        render(<MoviePage />);
        const addReviewButton = screen.getByRole("button", {
          name: /add review/i,
        });
        await userEvent.click(addReviewButton);
        const reviewTitle = await screen.findByRole("textbox", {
          name: /title/i,
        });
        expect(
          screen.queryByText(/title cannot be more than 40 characters/i)
        ).toBeNull();
        await userEvent.type(reviewTitle, dummyTitle);
        const submitButton = screen.getByRole("button", {
          name: /submit review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(
            screen.getByText(/title cannot be more than 40 characters/i)
          ).toBeInTheDocument();
        });
      });
    });

    describe("Description input tests", () => {
      it("Description is required", async () => {
        render(<MoviePage />);
        const addReviewButton = screen.getByRole("button", {
          name: /add review/i,
        });
        await userEvent.click(addReviewButton);

        expect(screen.queryByText(/description is required/i)).toBeNull();

        const submitButton = screen.getByRole("button", {
          name: /submit review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(
            screen.getByText(/description is required/i)
          ).toBeInTheDocument();
        });
      });

      it("Description cannot be longer than 200", async () => {
        render(<MoviePage />);
        const addReviewButton = screen.getByRole("button", {
          name: /add review/i,
        });
        await userEvent.click(addReviewButton);
        const reviewDescription = await screen.findByRole("textbox", {
          name: /description/i,
        });
        expect(
          screen.queryByText(/description cannot be more than 200 characters/i)
        ).toBeNull();
        await userEvent.type(reviewDescription, dummyDescription);
        const submitButton = screen.getByRole("button", {
          name: /submit review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(
            screen.getByText(/description cannot be more than 200 characters/i)
          ).toBeInTheDocument();
        });
      });
    });
  });

  describe("Edit review tests", () => {
    it("Edit review modal", async () => {
      render(<MoviePage />);
      const editReviewButton = await screen.findByRole("button", {
        name: /edit/i,
      });
      await userEvent.click(editReviewButton);
      await waitFor(() => {
        const editReviewTitle = screen.getByRole("heading", {
          name: /edit review/i,
        });
        expect(editReviewTitle).toBeInTheDocument();
      });
    });

    it("Submit edit review", async () => {
      render(<MoviePage />);
      const editReviewButton = screen.getByRole("button", {
        name: /edit/i,
      });
      await userEvent.click(editReviewButton);

      const reviewTitle = await screen.findByRole("textbox", {
        name: /title/i,
      });
      const reviewDescription = await screen.findByRole("textbox", {
        name: /description/i,
      });
      const reviewRating = await screen.findByLabelText("star-1");
      const submitButton = screen.getByRole("button", {
        name: /update review/i,
      });

      await userEvent.type(reviewTitle, "1234");
      await userEvent.type(reviewDescription, "1234");
      await userEvent.click(reviewRating);
      await userEvent.click(submitButton);

      await waitFor(() => {
        expect(mutationFn).toBeCalledTimes(1);
      });
    });

    describe("Title input tests", () => {
      it("Title is required", async () => {
        render(<MoviePage />);
        const editReviewButton = await screen.findByRole("button", {
          name: /edit/i,
        });
        await userEvent.click(editReviewButton);

        expect(screen.queryByText(/title is required/i)).toBeNull();

        const reviewTitle = await screen.findByRole("textbox", {
          name: /title/i,
        });

        await userEvent.clear(reviewTitle);

        const submitButton = screen.getByRole("button", {
          name: /update review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(screen.getByText(/title is required/i)).toBeInTheDocument();
        });
      });

      it("Title cannot be longer than 40", async () => {
        render(<MoviePage />);
        const editReviewButton = await screen.findByRole("button", {
          name: /edit/i,
        });
        await userEvent.click(editReviewButton);
        const reviewTitle = await screen.findByRole("textbox", {
          name: /title/i,
        });
        await userEvent.clear(reviewTitle);
        expect(
          screen.queryByText(/title cannot be more than 40 characters/i)
        ).toBeNull();
        await userEvent.type(reviewTitle, dummyTitle);
        const submitButton = screen.getByRole("button", {
          name: /update review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(
            screen.getByText(/title cannot be more than 40 characters/i)
          ).toBeInTheDocument();
        });
      });
    });

    describe("Description input tests", () => {
      it("Description is required", async () => {
        render(<MoviePage />);
        const editReviewButton = await screen.findByRole("button", {
          name: /edit/i,
        });
        await userEvent.click(editReviewButton);
        expect(screen.queryByText(/description is required/i)).toBeNull();

        const reviewDescription = await screen.findByRole("textbox", {
          name: /description/i,
        });
        await userEvent.clear(reviewDescription);

        const submitButton = screen.getByRole("button", {
          name: /update review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(
            screen.getByText(/description is required/i)
          ).toBeInTheDocument();
        });
      });

      it("Description cannot be longer than 200", async () => {
        render(<MoviePage />);
        const editReviewButton = await screen.findByRole("button", {
          name: /edit/i,
        });
        await userEvent.click(editReviewButton);
        const reviewDescription = await screen.findByRole("textbox", {
          name: /description/i,
        });
        expect(
          screen.queryByText(/description cannot be more than 200 characters/i)
        ).toBeNull();
        await userEvent.clear(reviewDescription);
        await userEvent.type(reviewDescription, dummyDescription);
        const submitButton = screen.getByRole("button", {
          name: /update review/i,
        });
        await userEvent.click(submitButton);

        await waitFor(() => {
          expect(
            screen.getByText(/description cannot be more than 200 characters/i)
          ).toBeInTheDocument();
        });
      });
    });
  });
});
