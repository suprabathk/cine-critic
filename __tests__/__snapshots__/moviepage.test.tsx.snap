// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moviepage tests Snapshot testing 1`] = `
[
  <div
    className="moviePageBanner"
    style={
      {
        "backgroundImage": "linear-gradient(rgba(0, 0, 0, 0.9), rgba(0, 0, 0, 0.9)), url(http://image.tmdb.org/t/p/w500/)",
      }
    }
  >
    <div
      className="MuiStack-root css-1hdc98z-MuiStack-root"
    >
      <img
        alt="poster"
        className="moviePagePoster"
        src="http://image.tmdb.org/t/p/w500/"
      />
      <div
        className="MuiStack-root css-r0zwdj-MuiStack-root"
      >
        <div
          className="MuiBox-root css-8atqhb"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4 css-1gdl256-MuiTypography-root"
          >
            title
          </h4>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2 css-7husxj-MuiTypography-root"
          >
            overview
          </h6>
          <div
            className="MuiStack-root css-ztx1wn-MuiStack-root"
          >
            <div
              className="MuiStack-root css-13dhmk0-MuiStack-root"
            />
            <div
              className="MuiStack-root css-1sgj6ch-MuiStack-root"
            >
              <div
                className="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-13htn8f-MuiChip-root"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarMonthOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zm0-12H5V6h14v2zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
                  />
                </svg>
                <span
                  className="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  release
                </span>
              </div>
              <div
                className="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-13htn8f-MuiChip-root"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
                  data-testid="AccessAlarmIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m22 5.72-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39 6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                  />
                </svg>
                <span
                  className="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  0 Mins
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiStack-root css-qdx1yi-MuiStack-root"
        >
          <span
            aria-label="0 Stars"
            className="MuiRating-root MuiRating-sizeMedium Mui-readOnly MuiRating-readOnly css-ryrseu-MuiRating-root"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="img"
          >
            <span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <div
            className="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-13htn8f-MuiChip-root"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="PeopleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiBox-root css-1um3uk5"
  >
    <div
      className="MuiStack-root css-bpo09q-MuiStack-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-79qqyf-MuiTypography-root"
      >
        Reviews
      </h5>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeSmall MuiButton-containedSizeSmall css-3wxg0j-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Add review
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiStack-root css-1gbhmed-MuiStack-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5f3tb7-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            className="MuiBox-root css-0"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5 css-fapfut-MuiTypography-root"
            >
              reviewTitle
            </h5>
            <p
              className="MuiTypography-root MuiTypography-subtitle2 css-h1ew73-MuiTypography-root"
            >
              reviewDescription
            </p>
          </div>
          <div
            className="MuiStack-root css-m2q5nn-MuiStack-root"
          >
            <div
              className="MuiStack-root css-14k604c-MuiStack-root"
            >
              <span
                aria-label="0 Stars"
                className="MuiRating-root MuiRating-sizeMedium Mui-readOnly MuiRating-readOnly css-ryrseu-MuiRating-root"
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
                role="img"
              >
                <span>
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
              <div
                className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
                <span
                  className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  reviewUsername
                </span>
              </div>
            </div>
            <div
              className="MuiStack-root css-1hv8ua0-MuiStack-root"
            >
              <div
                className="MuiBox-root css-1a5svn"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit MuiButton-fullWidth css-1dqom5d-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  Edit
                  <span
                    className="MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="EditIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                className="MuiBox-root css-1a5svn"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit MuiButton-fullWidth css-1dqom5d-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  Delete
                  <span
                    className="MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DeleteForeverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
